addons/l10n_latam_check/tests/test_third_party_checks.py

```diff
@@ -0,0 +1,172 @@
+                active_model='account.payment', active_ids=[check.id]).create({'destination_journal_id': bank_journal.id})._create_payments()
+
+        # Check Rejection
+        vals = {
+            'l10n_latam_check_id': check.id,
+            'amount': 1,
+            'payment_type': 'inbound',
+            'journal_id': self.rejected_check_journal.id,
+            'is_internal_transfer': True,
+            'payment_method_line_id': self.rejected_check_journal._get_available_payment_method_lines('inbound').filtered(lambda x: x.code == 'in_third_party_checks').id,
+            'destination_journal_id': bank_journal.id,
+        }
+        bank_rejection = self.env['account.payment'].create(vals)
+        bank_rejection.action_post()
+        self.assertEqual(check.l10n_latam_check_current_journal_id, self.rejected_check_journal, 'Current journal was not computed properly on return')
+        # check dont reject twice
+        with self.assertRaisesRegex(ValidationError, "it seems it has been moved by another payment"), self.cr.savepoint():
+            self.env['account.payment'].create(vals).action_post()
+
+        # Check Claim/Return to customer
+        vals = {
+            'l10n_latam_check_id': check.id,
+            'amount': 1,
+            'partner_id': self.partner_a.id,
+            'payment_type': 'outbound',
+            'journal_id': self.rejected_check_journal.id,
+            'payment_method_line_id': self.rejected_check_journal._get_available_payment_method_lines('outbound').filtered(lambda x: x.code == 'out_third_party_checks').id,
+        }
+        customer_return = self.env['account.payment'].create(vals)
+        customer_return.action_post()
+        self.assertFalse(check.l10n_latam_check_current_journal_id, 'Current journal was not computed properly on customer return')
+        # check dont return twice
+        with self.assertRaisesRegex(ValidationError, "it seems it has been moved by another payment"), self.cr.savepoint():
+            self.env['account.payment'].create(vals).action_post()
+
+        operations = self.env['account.payment'].search([('l10n_latam_check_id', '=', check.id), ('state', '=', 'posted')], order="date desc, id desc")
+        self.assertEqual(len(operations), 5, 'There should be 5 operations on the check')
+        self.assertEqual(operations[0], customer_return, 'Last operation should be customer return')
+        self.assertEqual(operations[2], bank_rejection, 'Previous operation should be supplier return')
+        self.assertEqual(operations[4], deposit, 'First operation should be customer delivery')
+
+    def test_04_check_transfer(self):
+        """ Test transfer between third party checks journals """
+        check = self.create_third_party_check()
+
+        # Transfer to rejected checks journal (usually is to another third party checks journal, but for test purpose is the same)
+        transfer1 = self.env['l10n_latam.payment.mass.transfer'].with_context(
```

Comment:
ci/style is complaining about unused variable

Reply:
removed!